src/screens/LiquidityPools/RemoveLiquidity.js

```diff
@@ -146,54 +147,53 @@ const AddLiquidityScreen = ({
     );
   }, [poolTokenAmount, obtainedTokenFieldsValid, poolTokenFieldValid]);
 
-  const onObtainedAssetValueChange = (newValue: string, tokenIndex: number) => {
-    const assetsAmounts = calculateProportionalRemoveLiquidityAssetValues(
-      pool,
-      parseFloat(newValue) || 0,
-      tokenIndex,
-      liquidityPoolsReducer,
-    );
+  const onPoolTokenAmountChange = (newValue: string) => {
+    const amounts = calculateProportionalAssetAmountsForRemoval(pool, newValue || '0', null, liquidityPoolsState);
+    setPoolTokenAmount(newValue);
+    setObtainedAssetsValues(amounts.pairTokens.map((amount) => formatAmount(amount)));
+  };
 
-    setPoolTokenAmount(formatAmount(assetsAmounts.pop()));
+  const onObtainedAssetValueChange = (tokenIndex: number, newValue: string, newPercent: number | void) => {
+    const amounts = calculateProportionalAssetAmountsForRemoval(pool, newValue || '0', tokenIndex, liquidityPoolsState);
 
-    const formattedAssetsValues = assetsAmounts
-      .map((amount, i) => i === tokenIndex ? newValue : formatAmount(amount));
-    setObtainedAssetsValues(formattedAssetsValues);
-  };
+    // Ensure we will withdraw everything if user want to remove max value.
+    if (newPercent === 100) {
+      onPoolTokenAmountChange(poolStats?.userLiquidityTokenBalance.toFixed() ?? '0');
```

Comment:
This `?? '0'` is to satisfy Flow null check and should happen only when `poolStats` are not available. IMO I would leave it as is, because there is no harm to the user.